src/MoBi.Presentation/Tasks/Edit/EditTaskFor.cs

```diff
@@ -66,7 +75,89 @@ public virtual void Edit(T objectToEdit)
       public virtual void Rename(T objectBase, IEnumerable<IObjectBase> existingObjectsInParent, IBuildingBlock buildingBlock)
       {
          var forbiddenNames = GetForbiddenNames(objectBase, existingObjectsInParent);
-         _context.AddToHistory(_interactionTask.Rename(objectBase, forbiddenNames, buildingBlock));
+         _context.AddToHistory(rename(objectBase, forbiddenNames, buildingBlock));
```

Comment:
Here now instead of calling _interactionTask, which calls ObjectBaseTask, we can just do all the renaming here because this is a generic class.

Reply:
what's the reason of doing this? You are adding dependencies in all derives class. There is probably a good reasons for this but I don't see it

Reply:
It could be done another way.

What we had before was an ObjectBaseTask that did only this renaming in the same way for all objects. That dependency was part of InteractionTask which was part of InteractionTaskContext.

These two general dependencies could go in InteractionTaskContext so that not so many classes have direct dependency on these objects, only InteractionTaskContext would have to change.

I'll move it around and see how that fits.

Reply:
It's nicer this way. The dependencies are still the dependencies, but now these ones are bundled into InteractionTaskContext instead of in the constructors of all these EditTasks.